[GeoMechanicsApplication] Addressed several code smells related to custom processes

avdg81 opened the pull request:
**📝 Description**
Addressed several code smells related to custom processes.

**🆕 Changelog**
Changes include:
- Made some `protected` data members `private`. Added getter functions when needed.
- Replaced some explicit type names by `auto`.
- Fixed a few typos.
- Improved the layout.

avdg81 committed:
Fixed more code smells found in `custom_processes`

- Declared some member functions `const`.
- Changed a parameter type to a reference-to-const.
- Improved the layout of some function arguments.
- Replaced a redundant type by `auto`.
- Removed an overridden member function since its body was identical to
  the base class implementation.
- Use a range-based for loop rather than an index to iterate a vector.

avdg81 committed:
Made seven `protected` data members `private`

Added getters to access the values of five data members.

Other changes include:
- Replaced some explicit type names by `auto`.
- Several minor style changes.
- Added a `static_cast` to avoid a difference in signedness.
- Corrected a typo.

avdg81 committed:
Initialize a pointer when it is declared

Also moved the pointer declaration closer to its usage and renamed it.

avdg81 committed:
Made 13 `protected` data members `private`

And getter functions for ten of them.

avdg81 committed:
Reformatted the changes using Clang-format

rfaasse commented on applications/GeoMechanicsApplication/custom_processes/apply_constant_boundary_hydrostatic_pressure_process.hpp:
```diff
@@ -46,7 +46,7 @@ class ApplyConstantBoundaryHydrostaticPressureProcess : public Process
             }  )" );
 
-        // Some values need to be mandatorily prescribed since no meaningful default value exist. For this reason try accessing to them
+        // Some values need to be mandatory prescribed since no meaningful default value exist. For this reason try accessing to them
```
Not too sure if this sentence is now completely correct. Maybe 'Some values are mandatory, since ...'? Or otherwise just remove 'mandatory' from the sentence

avdg81 replied:
Good catch. I have rephrased it to "Some values have to be input by the user since ..."

rfaasse commented on applications/GeoMechanicsApplication/custom_processes/apply_boundary_hydrostatic_pressure_table_process.hpp:
```diff
@@ -53,14 +53,14 @@ class ApplyBoundaryHydrostaticPressureTableProcess : public ApplyConstantBoundar
+        block_for_each(GetModelPart().Nodes(), [&deltaH, &var, this](Node& rNode) {
+            const double distance = GetReferenceCoordinate() - rNode.Coordinates()[GetGravityDirection()];
+            const double pressure               = GetSpecificWeight() * (distance + deltaH);
+            rNode.FastGetSolutionStepValue(var) = std::max(pressure, 0.0);
```
Might be github, but not sure if this is correctly formatted.

avdg81 replied:
I have attempted to reformat this block of code using Clang-format, but nothing changes here. Which part may not be correctly formatted in your opinion?

rfaasse replied:
Double checked, my mistake it aligns with line 63 instead of 61, the formatting is fine like this

rfaasse reviewed (dismissed):
Looks good, should get rid of quite some code smells (especially the 'no protected member' ones) I just have 2 minor comments, but looks good to go!

avdg81 committed:
Rephrased a comment

avdg81 committed:
Renamed several variables/function parameters

Some names were not compliant with the Kratos Style Guide. Those have
been changed.

In addition, some explicit type names have been replaced by `auto`.

avdg81 committed:
Rephrased a comment

WPK4FEM reviewed (approved):
Looks consistent to me
